latch/type_engine/__init__.py

```diff
@@ -0,0 +1,386 @@
+    _dataclass._from_dataclass = True
+    return _dataclass
+
+
+def _guess_python_class(literal_type: LiteralType):
+
+    # Dataclass JSON implementation
+    if literal_type.metadata is not None and "definitions" in literal_type.metadata:
+        schema_name = _get_schema_name(literal_type.metadata)
+        return _construct_class_from_json_schema(
+            literal_type.metadata["definitions"], schema_name
+        )
+    # TODO: Record implementation
+
+    raise ValueError(f"Unable to construct a python for literal_type: {literal_type}")
+
+
+###
+
+
+def guess_python_type(literal_type: LiteralType):
+    """Transform a LiteralType to a Python type."""
+
+    if literal_type.simple is not None:
+        if literal_type.simple == 9:
+            return _guess_python_class(literal_type)
+        return _simple_table[literal_type.simple]
+
+    if literal_type.collection_type is not None:
+        return typing.List[guess_python_type(literal_type.collection_type)]
+
+    if literal_type.blob is not None:
+
+        dim = literal_type.blob.dimensionality
+        if dim == 0:
+            return LatchFile
+        else:
+            return LatchDir
+
+    if literal_type.union_type is not None:
+
+        variant_types = [
+            guess_python_type(variant) for variant in literal_type.union_type.variants
+        ]
+
+        # Trying to directly construct set of types will throw error if list is
+        # included as 'list' is not hashable.
+        unique_variants = []
+        for t in variant_types:
+            if t not in unique_variants:
+                unique_variants.append(t)
+
+        return typing.Union[tuple(variant_types)]
+
+    if literal_type.enum_type is not None:
+
+        class _VariantCarrier(enum.Enum):
+            ...
+
+        _VariantCarrier._variants = literal_type.enum_type.values
+        # Construct a unique symbol to represent each enum as a python class.
+
+        global _enum_counter
+        _VariantCarrier._name = f"enum{_enum_counter}"
+        _enum_counter += 1
+
+        return _VariantCarrier
+
+    raise NotImplementedError(
+        f"The flyte literal_type {literal_type} cannot be transformed to a python type."
+    )
+
+
+def best_effort_python_val(t: typing.T):
+    """Produce a "best-effort" default value given a python type.
+
+    Unlike `guess_python_val` there is no Literal value acting as a guide.
+    """
+
+    if t in _primitive_table:
+        return _primitive_table[t]
+
+    if t is list:
+        return []
+
+    file_like_table = {
+        LatchDir: LatchDir("latch:///foobar"),
+        LatchFile: LatchFile("latch:///foobar"),
+    }
+    if t in file_like_table:
+        return file_like_table[t]
+
+    if type(t) is enum.EnumMeta:
+        return f"{t._name}.{t._variants[0]}"
+
+    if get_origin(t) is list:
+        list_args = get_args(t)
+        if len(list_args) == 0:
+            return []
+        return [best_effort_python_val(arg) for arg in list_args]
```

Comment:
why do this? why not just

```suggestion
    if get_origin(t) is list:
        return []
```

and get rid of the list if-clause at the top too

Reply:
because lists with more detail about what types they hold should generate more descriptive default values

```
>> best_effort_python_val(typing.List)
[]
>> best_effort_python_val(typing.List[str])
["foo"]
>> best_effort_python_val(typing.List[typing.List[str]])
[["foo]]
```